geopandas/geodataframe.py

```diff
@@ -575,6 +575,47 @@ def merge_geometries(block):
 
         return aggregated
 
+    # overrides GeoPandasBase method
+    def explode(self):
+        """    
+        Explode the geodataframe's muti-part geometries into single geometries.
+
+        Each row containing a multi-part geometry will be split into
+        multiple rows with single geometries, thereby increasing the vertical size
+        of the geodataframe. The index of the input geodataframe is no longer
+        unique and is replaced with a multi-index. 
+
+        The output geodataframe has an index based on two columns (multi-index) 
+        i.e. 'level_0' (index of input geodataframe) and 'level_1' which is a new
+        zero-based index for each single part geometry per multi-part geometry
+
+            
+        Returns
+        -------
+        geo_df (GeoDataFrame) : exploded geodataframe with each single geometry
+            as a separate entry in the geodataframe. The GeoDataFrame has a
+            multi-index set to columns level_0 and level_1.
+            
+        """
+        df_copy = self.copy() 
+        # for the case where the index is not named, set to equal level = 0
+        # so merge operation on index name can generalise
+        if not self.index.name:
+            df_copy.index.name = 'level_0'
+
+        exploded_geom = df_copy._get_geometry().explode().reset_index()
+        exploded_geom.rename(
+            columns={0:'geometry','level_0':df_copy.index.name}, inplace=True)
+        df = exploded_geom.merge(
+            df_copy.drop(df_copy._geometry_column_name, axis=1),
+            left_on=df_copy.index.name,
+            right_index=True)
```

Comment:
I think you can replace this with 

```
pd.concat([df.drop(df._geometry_column_name, axis=1), exploded_geom.reset_index(level=-1)], axis=1)
```

or (`df.drop(df._geometry_column_name, axis=1).join(exploded_geom.reset_index(level=-1))`